src/electrum.rs

```diff
@@ -360,7 +376,7 @@ impl Rpc {
     fn transaction_broadcast(&self, (tx_hex,): &(String,)) -> Result<Value> {
         let tx_bytes = Vec::from_hex(tx_hex).context("non-hex transaction")?;
         let tx = deserialize(&tx_bytes).context("invalid transaction")?;
-        let txid = self.daemon.broadcast(&tx)?;
+        let txid = self.tx_broadcaster.broadcast(&tx)?;
```

Comment:
Why use `Arc<Daemon>` instead of just `Daemon`? A `&Daemon` can be passed to the `broadcast` method here.